src/UI/src/Routes.tsx

```diff
@@ -1,24 +1,36 @@
-import React from "react";
+import React, { Suspense } from "react";
 import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
 import { NavigationBar } from "./components";
 import { ErrorBoundary } from "./shared/components/errorBoundary";
+import { useRoles } from "./shared/components/useRoles";
 import {
   Home as HomeView,
   MessageDetail as MessageDetailView,
   NotFound as NotFoundView,
+  NotAuthorized as NotAuthorizedView,
 } from "./views";
 
 export const Routing: React.FC = () => {
+  const hasRequiredRole = useRoles();
+
   return (
-    <Router>
-      <ErrorBoundary>
-        <NavigationBar />
-        <Routes>
-          <Route path="/" element={<HomeView />} />
-          <Route path="/message/:id" element={<MessageDetailView />} />
-          <Route path="*" element={<NotFoundView />} />
-        </Routes>
-      </ErrorBoundary>
-    </Router>
+    <Suspense>
+      <Router>
+        <ErrorBoundary>
+          {hasRequiredRole ? (
+            <>
```

Comment:
I would use a component here. So that you can make use of suspense. read the comment in 'useRoles'